x/application/keeper/msg_server_undelegate_from_gateway.go

```diff
@@ -49,6 +51,34 @@ func (k msgServer) UndelegateFromGateway(ctx context.Context, msg *types.MsgUnde
 	// Remove the gateway from the application's delegatee gateway public keys
 	foundApp.DelegateeGatewayAddresses = append(foundApp.DelegateeGatewayAddresses[:foundIdx], foundApp.DelegateeGatewayAddresses[foundIdx+1:]...)
 
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	currentBlock := sdkCtx.BlockHeight()
+	sessionEndHeight := uint64(sessionkeeper.GetSessionEndBlockHeight(currentBlock))
+
+	// TODO_INVESTIGATE: When there is no undelegation, the undelegations map is nil
```

Comment:
There are two cases: 

1. Prod
- Search for where we initialize it and make sure we provide a default value
2. Tests
- Search for all `.Application{` in the code ane make sure it's initialized